tests/delete/tests.py

```diff
@@ -175,7 +186,7 @@ def test_restrict_path_cascade_indirect(self):
         a.restrict.save()
         msg = (
             "Cannot delete some instances of model 'R' because they are "
-            "referenced through a restricted foreign key: 'A.restrict'."
+            "referenced through restricted foreign keys: 'A.restrict'."
```

Comment:
Previous message was incorrect, IMO. We tried to delete a `P` instance so we should get a message about `P` even if restricted foreign keys are from other models (`R` in this case). Restricted foreign keys can be reached from a long and complicated paths so I don't think we can do much to improve this message.